workshops/micro_bit_flappy_bird/README.md

```diff
@@ -0,0 +1,363 @@
+Learn more about micro:bit [here](https://microbit.org/get-started/user-guide/overview/).
+
+## Setup
+
+Open [Makecode](https://makecode.microbit.org/) and click on :heavy_plus_sign:`New Project`.
+
+![Setup](https://cloud-kifx081bh.vercel.app/3setup.png)
+
+Give it a name.
+
+![naming project](https://cloud-kifx081bh.vercel.app/4name_your_roject.png)
+
+You should see something like this. 
+
+![Project created](https://cloud-kifx081bh.vercel.app/5preview.png)
+
+Once you finish setting up, let's get going:rocket:
+
+## JavaScript
+
+By default, MakeCode open's a project with code blocks. At top of the window, switch to JavaScript.
+
+![JavaScript](https://cloud-kifx081bh.vercel.app/6javascript.png)
+
+The MakeCode programming environment uses JavaScript along with the Static [TypeScript](https://www.typescriptlang.org/) language. Don't worry about TypeScript. It is super similar to JavaScript. We will use most of the micro:bit’s built-in JavaScript functions. For example, the `showString()` function displays a string on the LED screen. It will scroll to the left if it’s bigger than the screen.
+
+Try displaying your name. Here's mine: 😊
+
+```js
+basic.showString("Giridhar!")
+```
+
+![name](https://cloud-kifx081bh.vercel.app/7name.gif)
+
+## Step-1
+
+Let's start with adding the bird. We just have 25 LEDs. Don't expect more. We have to assume a blinking LED as a bird😂
+
+Don't mind about the previously existing code. 
+
+Create a new LED sprite pointing to the right. A sprite is like a little LED creature you can tell what to do.
+
+```js
+let bird: game.LedSprite = null
+bird = game.createSprite(0, 2)
+bird.set(LedSpriteProperty.Blink, 300)
+```
+
+A variable `bird` is declared in TypeScript. The syntax of declaring a varible in TypeScript is: `let identifier:data-type = value;`
+
+The `game.createSprite(x,y)` creates a sprite(our bird) in position (x, y) on the 5x5 grid of LEDs. Then we set our bird to blink every 300 milliseconds.
+
+You can see our bird flying like this:point_down:
+
+![blinking LED](https://cloud-70aevdanf.vercel.app/0bird.gif)
+
+## Step-2
+
+Now let's create some controls.  We’ll control the bird by pressing the A button to go up or the B button to go down.
+
+Micro:bit has built-in [event handlers](https://makecode.microbit.org/reference/event-handler) that are associated with a particular event, such as "button A pressed".  `.onButtonPressed` handler works when button A or B is pressed, or A and B together. They are just like event listeners in JavaScript but with custom events.
+
+```js
+input.onButtonPressed(Button.A, function () {
+    bird.change(LedSpriteProperty.Y, -1)
+})
+input.onButtonPressed(Button.B, function () {
+    bird.change(LedSpriteProperty.Y, 1)
+})
+```
+
+Now when button any is pressed, the position of the bird should change by 1 unit(up or down).
+
+![controls](https://cloud-70aevdanf.vercel.app/1controls.gif)
+
+## Step-3
+
+Let's create some randomly generated obstacles for the bird to crash💥
```

Comment:
```suggestion
Let's create some randomly generated obstacles for the bird to crash 💥.
```

Reply:
please try to keep a space between the words and emojis so that it doesn't feel cluttered